build_tools/docker/get_image_name.py

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env python3
+
+# Copyright 2023 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+"""Command line tool to get the fully qualified image name given short name.
+
+Syntax:
+  ./build_tools/docker/get_image_name.py {short_name}
+
+Where {short_name} is the last name component of an image in prod_digests.txt
+(i.e. "base", "nvidia", etc).
+
+This is used both in tree and out of tree to get a image name and current
+version without adding fully referencing sha256 hashes, etc.
+"""
+
+from pathlib import Path
+import sys
+
+if __name__ == "__main__":
+  if len(sys.argv) != 2:
+    print("ERROR: Expected image short name", file=sys.stderr)
+    sys.exit(1)
+  short_name = sys.argv[1]
+  this_dir = Path(__file__).resolve().parent
+  with open(this_dir / "prod_digests.txt", "rt") as f:
```

Comment:
FYI, this can be slightly shorter with

```python
with open(Path(__file__).with_name("prod_digests.txt"), "rt") as f:
```

Not worth a commit to change this, but pathlib is fun and has a native way to say "give me a sibling file". You also could use `read_text()` for reading

```python
name = next(l.strip() for l in Path(__file__).with_name("prod_digests.txt").read_text().splitlines() if l.strip().startswith(f"gcr.io/iree-oss/{short_name}@", None)
```

:golfing:

(kidding about putting it all one line, but you get the idea. This is my idea of a good time)

Reply:
Thanks - I'm still reprogramming my brain towards pathlib. Grew up on os.path. I'm still not there that all of this is readability improvements...

One nit: in some unknown situations, __file__ can be a relative path and calling parent() on it directly (and I assume with_name()) has broken in the field on some Python versions/environments. I've developed muscle memory to be explicit and robust on things like that because the terse forms are nice for colab but hard to track for in the wild with all of the things that go wrong in practice.

Reply:
Yeah I agree it's not all improvements, but I'm starting to appreciate it more.

Wait how can file be a relative path? `with_name` is doing pure textual manipulation, so I guess a symlink could break it, but I don't see how a relative path could. I guess resolving first may still avoid some potential bugs, but I think `foo.parent / bar` is literally equivalent to `foo.with_name(bar)` if the path is in fact one to a file and not a directory

Reply:
idk. I just resolved an issue on another project where this appeared to happen. It wasn't on my machine and I couldn't repro. I just code defensively on this stuff and apply little trust.